pkg/cache/clusterqueue.go

```diff
@@ -96,7 +96,6 @@ type Cohort struct {
 	// RequestableResources equals to the sum of LendingLimit when feature LendingLimit enabled.
 	RequestableResources FlavorResourceQuantities
 	Usage                FlavorResourceQuantities
-	Lendable             map[corev1.ResourceName]int64
```

Comment:
An alternative to removing this could be to keep it up to date in the cache not only the snapshot. (update it when a cq is added removed or it's lending specs are changing).

Reply:
Yes, I thought about it. But I'm not sure that it will more optimised. First of all it's a lot of places where need to support this code so it's potential risk for bugs. And also it's just using only if fire sharing enabled.

Reply:
I agree that it might be better to maintain the map in the cache, but also doing this late in the release cycle could introduce bugs.

But, we can still keep the map.

There are a lot of calls to `.DominantResourceShare` during a preemption calculation, where the value of the map doesn't change.

in `dominantResourceShare`, you can do:

```golang
lendable := c.lendable // c.lendable is the map
if lendable == nil {
  lendable = c.Lendable()
```

Reply:
Done